fre/cmor/cmor_finder.py

```diff
@@ -13,97 +23,130 @@
                 print_var_content(table_config_file, opt_var_name)
 
     elif var_list is not None:
         fre_logger.info('opt_var_name is None, and var_list is not None, looking for many variables worth of info!')
         for var in var_list:
             for json_table_config in json_table_configs:
-                #fre_logger.info(f'attempting to open {json_table_config}')
-                with open( json_table_config, "r", encoding = "utf-8") as table_config_file:
-                    #fre_logger.info(f'    var = {var}, var_list[{var}]={var_list[var]}')
+                with open(json_table_config, "r", encoding="utf-8") as table_config_file:
                     print_var_content(table_config_file, str(var_list[var]))
     else:
         fre_logger.error('this line should be unreachable!!!')
         assert False
 
-    return
+def make_simple_varlist(dir_targ, output_variable_list):
+    """
+    Generates a JSON file containing a list of variables from NetCDF files in a specified directory.
+    This function searches for NetCDF files in the given directory (or a subdirectory "ts/monthly/5yr" if not already included),
+    extracts variable names from the filenames, and writes these variable names to a JSON file.
+
+    Args:
+        dir_targ (str): The target directory to search for NetCDF files.
+        output_variable_list (str): The path to the output JSON file where the variable list will be saved.
+
+    Returns:
+        None
+
+    Raises:
+        Logs errors if no files are found in the directory or if no files match the expected pattern.
+        Logs a warning if only one file is found matching the pattern.
+
+    Notes:
+        The function assumes that the filenames of the NetCDF files contain the variable name as the second-to-last component
+        when split by periods ('.') and a datetime string as the third-to-last component.
+    """
+    one_file = next(glob.iglob(os.path.join(dir_targ, "*.nc")), None)
+    if not one_file:
+        fre_logger.error("No files found in the directory.")
+        return
+
+    one_datetime = os.path.basename(one_file).split('.')[-3]
+
+    # Find all files in the directory that match the datetime component
+    files = glob.glob(os.path.join(dir_targ, f"*{one_datetime}*.nc"))
+
+    # Check if any files were found    
+    if not files:
+        fre_logger.error("No files found matching the pattern.")
+        return
+    elif len(files) == 1:
+        fre_logger.warning("Warning: Only one file found matching the pattern.")
+    else:
+        fre_logger.info("Files found with %s in the filename. Number of files: %d", one_datetime, len(files))
+
+    # Create a dictionary of variable names extracted from the filenames
+    var_list = {os.path.basename(file).split('.')[-2]: os.path.basename(file).split('.')[-2] for file in files}
+
+    # Write the variable list to the output JSON file
+    with open(output_variable_list, 'w') as f:
+        json.dump(var_list, f, indent=4)
```

Comment:
Nice! Processing all variables is handy even though we'll probably encourage an explicit list.

Since only a minority of variables are CMOR variables, most of the discovered variables will not be CMOR-izable.